android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/FileBroadcastService.java

```diff
@@ -0,0 +1,218 @@
+    }
+
+    @Override
+    public IBinder onBind(Intent intent) {
+        return null;
+    }
+
+    @Override
+    public int onStartCommand(Intent intent, int flags, int startId) {
+      if (mFileBroadcastServer != null) {
+        mFileBroadcastServer.stop();
+      }
+      String mType = intent.getStringExtra(MIME_TYPE_KEY);
+      Log.d(TAG, mType);
+      Uri mUri = Uri.parse(intent.getStringExtra(FILE_KEY));
+      Log.d(TAG, mUri.toString());
+      port = Utils.getWifiDirectPort(this);
+      try {
+        mFile = getBytes(getContentResolver().openInputStream(mUri));
+      } catch (FileNotFoundException e) {
+        Log.d(TAG, e.getMessage());
+      } catch (IOException e) {
+        Log.d(TAG, e.getMessage());
+      }
+      mNotificationManager = NotificationManagerCompat.from(this);
+      mFileBroadcastServer = new FileBroadcastServer(port, mType, mFile);
+      try {
+        mFileBroadcastServer.start();
+        createNotification();
+      } catch (IOException e) {
+        Log.d(TAG, e.getMessage());
+      }
+      sendBroadcast(new Intent("server"));
+      WifiP2pManager mManager = (WifiP2pManager) this.getSystemService(Context.WIFI_P2P_SERVICE);
+      WifiP2pManager.Channel mChannel = mManager.initialize(this, this.getMainLooper(), null);
+      changeWifiName();
+      mManager.discoverPeers(mChannel, new WifiP2pManager.ActionListener() {
+        @Override
+        public void onSuccess() {
+          Log.d(TAG, "discovering");
+        }
+
+        @Override
+        public void onFailure(int reasonCode) {
+          Log.d(TAG, "discovery failed " + reasonCode);
+        }
+      });
+      return START_STICKY;
+    }
+
+
+
+    @Override
+    public void onDestroy() {
+      Log.d(TAG, "SERVICE onDestroy");
+      unregisterReceiver(stopServiceReceiver);
+      mFileBroadcastServer.stop();
+      mNotificationManager.cancel(BROADCASTING_NOTIFICATION_ID);
+      super.onDestroy();
+    }
+
+        // Surface a notification to the user that a URL is being broadcast
+    // The notification specifies the URL being broadcast (the long URL)
+    // and cannot be swiped away
+    private void createNotification() {
+      Intent resultIntent = new Intent();
+      TaskStackBuilder stackBuilder = TaskStackBuilder.create(this);
+      stackBuilder.addParentStack(BroadcastActivity.class);
+      stackBuilder.addNextIntent(resultIntent);
+      PendingIntent resultPendingIntent = stackBuilder.getPendingIntent(0,
+          PendingIntent.FLAG_UPDATE_CURRENT);
+      registerReceiver(stopServiceReceiver, new IntentFilter("myFilter2"));
+      PendingIntent pIntent = PendingIntent.getBroadcast(this, 0, new Intent("myFilter2"),
+          PendingIntent.FLAG_UPDATE_CURRENT);
+      NotificationCompat.Builder mBuilder = new NotificationCompat.Builder(this)
+        .setSmallIcon(R.drawable.ic_leak_add_white_24dp)
+        .setContentTitle("Physical Web is sharing with WifiDirect")
+        .setContentText("" + port)
```

Comment:
use Integer.toString(port), doing it this way works but creates an unnecessary StringBuilder